crates/ide/src/syntax_highlighting.rs

```diff
@@ -371,29 +373,25 @@ fn traverse(
         // string highlight injections, note this does not use the descended element as proc-macros
         // can rewrite string literals which invalidates our indices
         if let (Some(token), Some(descended_token)) = (token, descended_element.as_token()) {
-            let string = ast::String::cast(token);
-            let string_to_highlight = ast::String::cast(descended_token.clone());
-            if let Some((string, expanded_string)) = string.zip(string_to_highlight) {
-                if string.is_raw() {
-                    if inject::ra_fixture(hl, sema, &string, &expanded_string).is_some() {
-                        continue;
+            if ast::String::can_cast(token.kind()) && ast::String::can_cast(descended_token.kind())
```

Comment:
There is no need to do this extra check here given the  `if let Some((string, expanded_string)) = string.zip(string_to_highlight)` afterwards is checking the same with the casts already

Reply:
It seems that we need to clone `token` if we omit this line. I'm not sure this is acceptable.

Reply:
It would be fine to clone as cloning tokens is just a ref count increase, but we would do that a lot here so I guess its fine to leave it like this as well 👍